app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -145,10 +149,16 @@ def handle_bad_request_error(err, enrollment)
   end
 
   def handle_standard_error(err, enrollment)
+    response_attributes = if err.respond_to?(:response)
```

Comment:
I'm trying to understand when a `StandardError` would ever respond to the `response` method? Seems to me that may have been the issue in the first place? I also don't see any test cases which exercise this.

Would an alternative be to exclude `**response_attributes` from the analytics logged in this method, and keep the addition of the NewRelic agent call?

Reply:
Despite my prior comment, based on the errors being logged in NewRelic, it would seem that the issue isn't so much whether `error` responds to `response`, but that `response` is explicitly `nil` when `response_analytics_attributes` tries to pick values from the response†. I think this would be accounted for in the revisions proposed here (checking `response.present?`), but then it's not clear what scenarios we're expecting or not expecting `error.response` to exist at all.

---

† The reason being that the exceptions are occurring inside `response_analytics_attributes` (`undefined method '[]' for nil:NilClass`), whereas if `error` didn't respond to the `response`, I would expect the exception to occur in this method as `undefined method 'response'`.

Reply:
Ok so digging a bit more, I think the errors we're handling here are other Faraday exceptions besides [the `Faraday::BadRequestError` we're currently handling](https://github.com/18F/identity-idp/blob/69ac7fd6fbde03cdf38127f60feebd1615c01e74/app/jobs/get_usps_proofing_results_job.rb#L108), such as `Faraday::ServerError` ([related test case](https://github.com/18F/identity-idp/blob/69ac7fd6fbde03cdf38127f60feebd1615c01e74/spec/services/usps_in_person_proofing/proofer_spec.rb#L159-L169)), which _extend_ `StandardError` and thus are caught in this path and respond to `response`?

I think that helps clarify what's actually happening, though (separately) we may want to revisit how we're dealing with Faraday-specific errors, if we're handling some but not all explicitly.

Reply:
Yeah that would be nice for observability too. I expect we'll see occasional Faraday errors and it won't be out of the ordinary. Non-Faraday standard errors should probably be looked at each time though

Reply:
Ticket to address this thread: https://cm-jira.usa.gov/browse/LG-7706